utils.py

```diff
@@ -48,6 +51,28 @@ def perform_rmd_conversion(stream):
             out_file.close()
             os.remove(out_path)
         return read_outfile
+    
+# We need to sanitize any code blocks (ex ```{bash}) from the rmd before rendering it
+# This is because kitr (https://rmarkdown.rstudio.com/authoring_quick_tour.html#Rendering_Output.), which powers our rendering
+# Actually executes any code in a labeled codeblock. By processing out any of these labels, we ensure we display a preview without a possible arbitrary code execution vulnerability
+# See this document, issue number one, for more details on the vulnerability: https://docs.google.com/document/d/1aNCOKitTJH-GEkBSR4i-x91O0OQCZ8ZYa3feXtkja94/edit#heading=h.rvpr6zoz0jem 
+# Takes a string and returns binary
+def sanitize_rmd(data):
+    print('printing data to sanitize pre-split', data)
+    lines = data.split('\n')
+    print('printing lines of file', lines)
+    sanitized_file = []
+    for line in lines:
+        if line.startswith('```'):
```

Comment:
i thought of doing this in replace with regex, but this seemed a bit brittle. Rather, I just look for any lines starting with the MD-specific-syntax of '```', and process out any code directive that could be there (see knitr docs linked in comment above function for more details)